src/utils.rs

```diff
@@ -80,6 +81,19 @@ pub(crate) fn connect_channel_lazily(uri_string: &str) -> Result<Channel, Channe
     Ok(endpoint.connect_lazy())
 }
 
+pub(crate) fn connect_channel_lazily_configurable(
```

Comment:
Having a general utils class feels strange to me as well. I think we should eventually make a dedicated place for internal grpc logic that isn't client-specific to live.